project_classification/project_classification_view.xml

```diff
@@ -46,10 +48,10 @@
       <field name="model">project.project</field>
       <field name="inherit_id" ref="project.edit_project" />
       <field name="arch" type="xml">
-        <xpath expr="/form/sheet/notebook/page[@string='Other Info']/group/group[3]/field[@name='parent_id']" position="replace">
-          <field name="classification_id" widget="selection" on_change="onchange_classification_id(classification_id)"/>
+        <xpath expr="/form/sheet/notebook/page[@string='Other Info']//field[@name='parent_id']" position="replace">
+          <field name="classification_id" required="1" widget="selection" on_change="onchange_classification_id(classification_id)"/>
```

Comment:
Agree with @yvaucher.
An idea: a migration script that:
* returns there is no records or all records have a classification_id
* create the column classification_id as nullable
* create a classification named "unclassed" or something like that
* update classification_id with the new ID
* set classification_id NOT NULL